tools/formatting/format-check.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def check(base_commit):
+    try:
+        p = subprocess.Popen(
+                [
+                    "python",
+                    os.path.join("tools", "formatting", "git-clang-format.py"),
```

Comment:
Why do you need to start python from python? May be it would be better to do everything in one script here?

Reply:
Because `git-clang-format.py` comes from llvm and I don't want to touch it. In fact I would like to updated on a later PR since there is a new version available. I tried to go the path of importing `git-clang-format.py` but unfortunately this is not good for importing as it might `exit()`.